site/layouts/_default/docs.html

```diff
@@ -18,12 +18,18 @@ <h1 class="bd-title" id="content">{{ .Title | markdownify }}</h1>
 
       {{ if (eq .Page.Params.toc true) }}
         <div class="bd-toc mt-4 mb-5 my-md-0 ps-xl-3 mb-lg-5 text-muted">
-          <strong class="d-block h6 my-2 pb-2 border-bottom">On this page</strong>
-          {{ .TableOfContents }}
+          <button class="btn btn-link link-dark p-md-0 mb-2 mb-md-0 fw-semibold text-decoration-none bd-toc-toggle" type="button" data-bs-toggle="collapse" data-bs-target="#tocContents" aria-expanded="false" aria-controls="tocContents">
```

Comment:
Here it is: I'd go with `.d-lg-none` on the button itself, and another element (`p` or `hn` most likely) with `.d-none.d-lg-block`. WDYT?